cirq-core/cirq/sim/clifford/act_on_stabilizer_ch_form_args.py

```diff
@@ -47,8 +55,13 @@ def __init__(
                 effects.
             log_of_measurement_results: A mutable object that measurements are
                 being recorded into.
+            initial_state: The initial state for the simulation in the
+                computational basis. Represented as a big endian int.
         """
         super().__init__(prng, qubits, log_of_measurement_results)
+        if state is None:
+            qubit_map = {q: i for i, q in enumerate(self.qubits)}
+            state = clifford_simulator.CliffordState(qubit_map, initial_state=initial_state).ch_form
         self.state = state
```

Comment:
Can we do something like

```python

    @_compat.deprecated_parameter(
        deadline='v0.15',
        fix='Specify all the arguments with keywords, use initial_state instead of state.',
        parameter_desc='positional arguments',
        match=lambda args, kwargs: len(args) != 1 or 'state' in kwargs,
    )
    def __init__(
        self,
        state: Optional['cirq.StabilizerStateChForm'] = None,
        prng: Optional[np.random.RandomState] = None,
        log_of_measurement_results: Optional[Dict[str, Any]] = None,
        qubits: Optional[Sequence['cirq.Qid']] = None,
        initial_state: Union[int, 'cirq.StabilizerStateChForm'] = 0,
    ):
        """Initializes with the given state and the axes for the operation.

        Args:
            state: The StabilizerStateChForm to act on. Operations are expected
                to perform inplace edits of this object.
            qubits: Determines the canonical ordering of the qubits. This
                is often used in specifying the initial state, i.e. the
                ordering of the computational basis states.
            prng: The pseudo random number generator to use for probabilistic
                effects.
            log_of_measurement_results: A mutable object that measurements are
                being recorded into.
            initial_state: The initial state for the simulation in the
                computational basis. Represented as a big endian int.
        """
        super().__init__(prng, qubits, log_of_measurement_results)
        initial_state = state or initial_state
        if isinstance(initial_state, int):
            qubit_map = {q: i for i, q in enumerate(self.qubits)}
            initial_state = clifford_simulator.CliffordState(qubit_map, initial_state=initial_state).ch_form
        self.state = initial_state
```

Then update all the callers to use `initial_state` instead of `state`? Or did you see a problem with that approach? If we can do that then everything else looks good.

Reply:
I see. You want to merge `state` into `initial_state`. I will update the code soon.